src/cdk-experimental/listbox/listbox.spec.ts

```diff
@@ -308,8 +308,246 @@ describe('CdkOption', () => {
       expect(listboxInstance._listKeyManager.activeItem).toEqual(optionInstances[2]);
       expect(listboxInstance._listKeyManager.activeItemIndex).toBe(2);
     });
+
+    it('should select all options using the select all method', () => {
+      let selectedOptions = optionInstances.filter(option => option.selected);
+
+      expect(selectedOptions.length).toBe(0);
+      expect(optionElements[0].hasAttribute('aria-selected')).toBeFalse();
+      expect(optionInstances[0].selected).toBeFalse();
+      expect(fixture.componentInstance.changedOption).toBeUndefined();
+
+      listboxInstance.setAllSelected(true);
+      fixture.detectChanges();
+
+      selectedOptions = optionInstances.filter(option => option.selected);
+      expect(selectedOptions.length).toBe(4);
+
+      for (const option of optionElements) {
+        expect(option.getAttribute('aria-selected')).toBe('true');
+      }
+
+      expect(fixture.componentInstance.changedOption).toBeDefined();
```

Comment:
I typically avoid toBeDefined since it returns true for "null", which can be a little confusing. ".not.toBe(undefined);" is less mistakable. May not matter for this usage, up to you.

Reply:
Since I am checking `changedOption.id` directly after, do you think it is even necessary to check that `changedOption` is defined?